### PR TITLE
Feature/drink edit and delete

### PR DESCRIPTION
**Pull request by eccard**

### CODE REVIEW

File: app/src/main/java/com/r5k/contacerveja/ui/drink/DrinkPagerAdapter.kt [hunk 1]
```diff
@@ -29,4 +29,25 @@ class DrinkPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
...
+                drinksList!!.add(i,drink)
+                notifyDataSetChanged()
+                break
+            }
+        }
+    }
+
+    fun removeDrinkFragment(drink: Drink){
+        for ( i in 0..drinksList!!.size) {
+            if (drinksList!![i].id == drink.id){
```
Comment (lssalgado) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/DrinkPagerAdapter.kt]: Extract to a function. 
`findDrinkIndexById(id : Int) : Int`

Comment (eccard) [reply app/src/main/java/com/r5k/contacerveja/ui/drink/DrinkPagerAdapter.kt]: done

**Review by leomsaippa — APPROVED** [review]

File: app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt [hunk 2]
```diff
@@ -78,4 +81,87 @@ class DrinkFragment : BaseFragment(),DrinkMVPView, View.OnClickListener {
...
         }
     }
+
+    private fun showEditDialog(){
+
+        val builder = mContext?.let { AlertDialog.Builder(it) }
+
+        // https://stackoverflow.com/questions/10695103/creating-custom-alertdialog-what-is-the-root-view
+        // Seems ok to inflate view with null rootView
+        val view = layoutInflater.inflate(com.r5k.contacerveja.R.layout.dialog_new_drink, null)
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt]: this package reference isn't necessary. 
```suggestion
        val view = layoutInflater.inflate(R.layout.dialog_new_drink, null)
```

File: app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt [hunk 3]
```diff
@@ -78,4 +81,87 @@ class DrinkFragment : BaseFragment(),DrinkMVPView, View.OnClickListener {
...
+
+        val newDrinkEditText = view.findViewById(com.r5k.contacerveja.R.id.edt_new_drink_name) as TextInputEditText
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt]: same here

File: app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt [hunk 4]
```diff
@@ -78,4 +81,87 @@ class DrinkFragment : BaseFragment(),DrinkMVPView, View.OnClickListener {
...
+        newDrinkEditText.setText(mDrink!!.name)
+
+        builder!!.setView(view)
+
+        // set up the ok button
+        builder.setPositiveButton(android.R.string.ok) { dialog, p1 ->
+            val newDrinkName = newDrinkEditText.text
+            var isValid = true
+            if (newDrinkName!!.isBlank()) {
+                newDrinkEditText.error = getString(com.r5k.contacerveja.R.string.validation_empty)
```
Comment (m4theuslim4) [inline app/src/main/java/com/r5k/contacerveja/ui/drink/view/DrinkFragment.kt]: same above

**Review by lssalgado — APPROVED** [review]

**Review by m4theuslim4 — APPROVED** [review]

**Pull request merged**